dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -1365,23 +1364,14 @@ public void publishWorkflowAndTestDiscourseTopicCreation() {
         final Workflow workflowByPathGithub = userWorkflowsApi.getWorkflowByPath(DOCKSTORE_TEST_USER2_GDC_DNASEQ_CWL_WORKFLOW + "/" + workflowName, null);
         final Workflow workflow = userWorkflowsApi.refresh(workflowByPathGithub.getId());
 
-        // Publish workflow
+        // Publish workflow, which will also add a topic
         userWorkflowsApi.publish(workflow.getId(), new PublishRequest(){
             public Boolean isPublish() { return true;}
         });
 
-        // Should not be able to create a topic in category 3 (for example)
-        Integer wrongCategoryId = 3;
+        // Should not be able to create a topic for the same workflow
         try {
-            curatorEntriesApi.setDiscourseTopic(workflow.getId(), wrongCategoryId);
-            fail("Should not be able to set discourse topic.");
-        } catch (ApiException ex) {
-        }
-
-        // Should not be able to create a topic for the same workflow in category 9
-        Integer correctCategoryId = 9;
-        try {
-            curatorEntriesApi.setDiscourseTopic(workflow.getId(), correctCategoryId);
+            curatorEntriesApi.setDiscourseTopic(workflow.getId());
```

Comment:
It wasn't clear to me, is the only time `setDiscourseTopic` set to throw exceptions for testing?
If so, I'd prefer to simplify and have it never throw exceptions but have the test look for the logging error message as per Slack

Reply:
It only throws exceptions in the case where you directly call the endpoint to add a topic, and not if the topic is being added through the publish endpoint. This is because I was worried it was bad practice to have an endpoint fail but return success.